amplify/backend/function/fetchDomainBalance/src/index.js

```diff
@@ -0,0 +1,61 @@
+require('cross-fetch/polyfill');
+
+const {
+    utils: { Logger },
+} = require('ethers');
+const { getPeriodFromNow, getDaysFromNow } = require('./utils');
+const ExchangeRatesFactory = require('./config/exchangeRates');
+const { getInOutActions, getTokensDatesMap, filterActionsWithinTimeframe, updatePeriodBalance, update30DaysBalance } = require('./services/actions');
+const { getTotalFiatAmountFor } = require('./services/tokens');
+Logger.setLogLevel(Logger.levels.ERROR);
+
+// @TODO maybe rename the lambda function
+
+/**
+ * @type {import('@types/aws-lambda').APIGatewayProxyHandler}
+ */
+exports.handler = async (event) => {
+    try {
+        const { colonyAddress, domainAddress, chainId, selectedCurrency, timeframePeriod = 4, timeframeType } = event.arguments?.input || {};
+        const periodFromNow = getPeriodFromNow(timeframePeriod, timeframeType);
+        const last30Days = getDaysFromNow(30);
+
+        const inOutActions = await getInOutActions(colonyAddress, domainAddress);
+        const inOutActionsWithinTimeframe = filterActionsWithinTimeframe(inOutActions, periodFromNow);
+        const periodBalance = updatePeriodBalance(timeframeType, timeframePeriod, inOutActionsWithinTimeframe, domainAddress);
+
+        const last30DaysActions = filterActionsWithinTimeframe(inOutActions, last30Days);
+        const last30DaysBalance = update30DaysBalance(last30DaysActions, domainAddress);
+
+        const exchangeRates = await ExchangeRatesFactory.getExchangeRates(getTokensDatesMap(inOutActionsWithinTimeframe), selectedCurrency, chainId);
+
+        const inOutPeriodBalance = {};
+
+        for (let period in periodBalance) {
+            const balance = periodBalance[period];
+            const totalIn = await getTotalFiatAmountFor(balance.in, exchangeRates);
+            const totalOut = await getTotalFiatAmountFor(balance.out, exchangeRates);
+            inOutPeriodBalance[period] = {
+                totalIn,
+                totalOut,
+            };
+        }
+
+        const last30DaysBalanceTotalIn = await getTotalFiatAmountFor(last30DaysBalance.in, exchangeRates);
+        const last30DaysBalanceTotalOut = await getTotalFiatAmountFor(last30DaysBalance.out, exchangeRates);
```

Comment:
Would be nice if this was configurable and not "hardcoded".

Ie: you give the lambda the number of days as an input variable, and it fetches amounts based on that, rather than always fetching it for 30 days

Reply:
@rdig you're right! Given how the inputs for this lambda are defined, we can easily make a request for the last 30 days 🌟 will tackle this soon